skimage/filters/_sparse.py

```diff
@@ -0,0 +1,69 @@
+import numpy as np
+from ._sparse_cy import _correlate_sparse_offsets
+
+
+def broadcast_mgrid(arrays):
```

Comment:
This definitely needs a docstring with examples. I wrote this and reading the code I can't figure out what it does. 😂 There's a lesson here that everyone needs to learn multiple times in their coding life. 😂 

iirc it's like `np.meshgrid(index='ij', sparse=True)`, but broadcasting the resulting arrays together.